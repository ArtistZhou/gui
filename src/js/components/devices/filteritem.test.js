import React from 'react';
import renderer from 'react-test-renderer';
import FilterItem from './filteritem';

describe('FilterItem Component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<FilterItem filterAttributes={['test']} filter={{ key: 'testkey', value: 'testvalue' }} filters={[{ key: 'testkey', value: 'testvalue' }]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
