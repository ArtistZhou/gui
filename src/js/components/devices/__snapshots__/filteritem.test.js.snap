// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterItem Component renders correctly 1`] = `
<div
  className="flexbox"
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <div
    className="margin-right"
  >
    Device matching:
  </div>
  <div
    aria-expanded={false}
    aria-owns={null}
    className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="combobox"
  >
    <div
      className="MuiFormControl-root MuiTextField-root"
      style={
        Object {
          "marginBottom": 15,
          "marginTop": 0,
        }
      }
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
        htmlFor="filter-selection-undefined"
        id="filter-selection-undefined-label"
      >
        Attribute
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
        onClick={[Function]}
      >
        <input
          aria-activedescendant={null}
          aria-autocomplete="both"
          aria-controls={null}
          aria-invalid={false}
          autoCapitalize="none"
          autoComplete="off"
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
          disabled={false}
          id="filter-selection-undefined"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          required={false}
          spellCheck="false"
          type="text"
          value=""
        />
        <div
          className="MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Clear"
            className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Clear"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="margin-left-small margin-right"
  >
    Equals
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root"
    style={
      Object {
        "marginBottom": 15,
        "marginTop": 0,
      }
    }
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink={true}
    >
      Value
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="testvalue"
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiIconButton-root margin-left MuiIconButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
    </span>
  </button>
  <div
    className="loaderContainer shrunk"
  >
    <div
      className="loader"
    >
      <span
        className="dot dot_1"
      />
      <span
        className="dot dot_2"
      />
      <span
        className="dot dot_3"
      />
      <span
        className="dot dot_4"
      />
    </div>
  </div>
</div>
`;
